compiler/optimizer/InductionVariable.cpp

```diff
@@ -3709,20 +3714,11 @@ bool TR_LoopStrider::reassociateAndHoistComputations(TR::Block *loopInvariantBlo
 
          TR::SymbolReference *internalPointerSymRef = (*_reassociatedAutos)[originalInternalPointerSymbol];
          originalNode->getFirstChild()->recursivelyDecReferenceCount();
-         node->decReferenceCount();
-         _reassociatedNodes.add(node);
```

Comment:
Why do we skip adding to `_reassociatedNodes` ?

Reply:
We aren't making any changes to array element offset computation (`1n4`). We simply hoist the dataAddr pointer node (`1n2`) and leave the offset node as is, so we don't need to add anything to `_reassociatedNodes`. Whereas in gencon, we need to update `_reassociatedNodes` because we hoist array header size (`2n5`) from offset node (`2n3`) along with array object load.

```
      off-heap mode:
      1n1   aladd (internal pointer)
      1n2      contiguousArrayDataAddrFieldSymbol (dataAddrPointer, internal pointer)
      1n3         array_obj (pinning array pointer)
      1n4      mul/shift/integer offset

      non off-heap mode:
      2n1   aladd (internal pointer)
      2n2      array_obj (pinning array pointer)
      2n3      add/sub offset
      2n4         index
      2n5         header_size/-header_size
```